### PR TITLE
0.5.0

### PR DESCRIPTION
**Pull request by carlvitzthum**
Big release that changes the functionality of hic2cool pretty significantly. 
- Commands are now run with `hic2cool <mode>`.
- Modes are `convert`, `update`, and `extract-norms`. The last of those replaces the `hic2cool_extractnorms.py` script
- `convert` not does NOT invert hic normalization vectors. Missing values in vectors are represented as `NaN`s
- `update` can be used to un-invert cooler weights from hic2cool files made from previous versions. File metadata is used to keep track of software versions and what has/hasn't been updated.
- `extract-norms` has been refactored for speed and usability within the greater hic2cool framework
- commands now have much better help information
- Lots of new tests
- README update that should cover a lot of the important stuff

### CODE REVIEW

File: hic2cool/hic2cool_utils.py [hunk 1]
```diff
@@ -736,41 +743,128 @@ def write_zooms_for_higlass(h5res):
             higlass_compat = False
             break
     if not higlass_compat:
-        print('WARNING: This hic file is not higlass compatible! Will not add [max-zoom] attribute.')
+        print('!!! WARNING: This hic file is not higlass compatible! Will not add [max-zoom] attribute.')
```
Comment (nvictus) [inline hic2cool/hic2cool_utils.py]: You could print logging messages to `sys.stderr`.

Comment (nvictus) [reply hic2cool/hic2cool_utils.py]: Not a huge deal though, since there is currently no meaningful stdout output.

Comment (carlvitzthum) [reply hic2cool/hic2cool_utils.py]: Good call. Error messages are printed to sys.stderr, so I might as well do the same for warning messages

Comment (nvictus) [reply hic2cool/hic2cool_utils.py]: You might want to define a logging function to reduce the noise of appending `file=sys.stderr` everywhere. There is, of course, the `logging` module too, but it's kind of overkill.

Comment (carlvitzthum) [reply hic2cool/hic2cool_utils.py]: Agreed, thanks for the feedback. Defined a logging method and also added some tests

**Pull request merged**